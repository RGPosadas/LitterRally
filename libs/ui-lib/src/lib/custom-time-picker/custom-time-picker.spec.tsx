import React, { ChangeEvent } from 'react';
import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import CustomTimePicker, { CustomTimePickerProps } from './custom-time-picker';

const mockOnChange: (e: ChangeEvent<HTMLInputElement>) => void = jest.fn();
const mockCustomTimePickerProps: () => CustomTimePickerProps = () => ({
  label: 'Text',
  value: '',
  fieldName: 'text',
  errorMessage: 'Text is not valid.',
  onChange: mockOnChange,
  hasErrors: true,
});

describe('CustomTimePicker', () => {
  it('should render successfully', () => {
    const props = mockCustomTimePickerProps();
    const { baseElement } = render(<CustomTimePicker {...props} />);
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', () => {
    const props = mockCustomTimePickerProps();
    const tree = renderer.create(<CustomTimePicker {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should have the correct time input', () => {
    const props = mockCustomTimePickerProps();
    const inputTime = '04:20';
    const utils = render(<CustomTimePicker {...props} />);
    const input = utils.getByTestId('custom-time-input');
    fireEvent.change(input, { target: { value: inputTime } });
    props.value = inputTime;
    utils.rerender(<CustomTimePicker {...props} />);
    expect(input.value).toBe(inputTime);
  });
});
