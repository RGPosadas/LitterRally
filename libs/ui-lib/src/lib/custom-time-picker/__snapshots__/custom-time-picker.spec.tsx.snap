// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTimePicker should match snapshot 1`] = `
<div
  className="custom-time-picker-container"
>
  <label
    className="custom-time-picker-label"
    htmlFor="text"
  >
    Text
  </label>
  <input
    className="custom-time-picker"
    data-testid="custom-time-input"
    id="text"
    name="text"
    onChange={[MockFunction]}
    type="time"
    value=""
  />
  <span
    className="error-message"
  >
    Text is not valid.
  </span>
</div>
`;
